test/parallel/test-http2-methods.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const h2 = require('http2');
+
+const server = h2.createServer();
+
+const methods = [undefined, 'GET', 'POST', 'PATCH', 'FOO', 'A B C'];
```

Comment:
Quote: http://httpwg.org/specs/rfc7231.html#method.overview

> The method token is case-sensitive because it might be used as a gateway to object-based systems with case-sensitive method names.

Add a lower- or mixed case method value.

> All general-purpose servers MUST support the methods GET and HEAD. All other methods are OPTIONAL.

Add HEAD as a method value.

Quote: http://httpwg.org/specs/rfc7230.html#rule.token.separators

> ```
>   token          = 1*tchar
> 
>   tchar          = "!" / "#" / "$" / "%" / "&" / "'" / "*"
>                  / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~" 
>                  / DIGIT / ALPHA
>                  ; any VCHAR, except delimiters
> ```

Whitespace is not allowed so `A B C` would be invalid.